src/libs/routes/authMiddleware.ts

```diff
@@ -0,0 +1,31 @@
+import  * as jwt from 'jsonwebtoken';
+import { hasPermission } from '../../libs/permission';
+import { permissions } from '../../libs/constants';
+import { error } from 'console';
+
+export default (module, permissionType ) => (req, res, next) => {
+  try {
+
+console.log('The config is', module, permissionType);
+console.log('Header is', req.headers.authorization);
+const token = req.headers.authorization;
+
+const decodedUser = jwt.verify(token, 'qwertyuiopasdfghjklzxcvbnm123456');
+console.log('User', decodedUser);
+if (hasPermission(permissions.getUser, decodedUser.role, permissionType)) {
```

Comment:
You are implicitly passing permission object  ( `permissions.getUser`)   in hasPermission() function this is not the right approach what if you have one more module in a permissions object.  
You should pass moduleName as string from here and hasPermission() function should be change accordingly